web-ng/src/app/shared/converters/firebase-data-converter.ts

```diff
@@ -367,8 +367,9 @@ export class FirebaseDataConverter {
    * @param data the source data in a dictionary keyed by string.
    */
   static toFeature(id: string, data: DocumentData): Feature | undefined {
-    if (!this.isFeatureValid(data)) {
-      console.warn(`Invalid feature ${id} in remote data store ignored`);
+    const errors = this.validateFeature(data);
```

Comment:
Hi @parulraheja98, I think extracting errors first into a data structure is probably overkill for this case. I was actually suggesting something different: just check the values as you extract them, and throw an exception when the value can't be converted. Would this work?

Reply:
I thought it would be much cleaner having an object with key value pair depicting the error :), but I will implement your approach :)

Reply:
That assumes we want to keep track of and report multiple issues with the local db. Since they should be a rare edge case, the extra complexity of keeping track of error might not be warranted. The key recommendation is to keep validation and conversion logic together so that they don't drift or so we don't need to look in two places to update conversion logic.